### ISSUE TITLE
WHERE functionalities are slightly changed.

### ISSUE THREAD

**Issue by wangmir** [huandu/go-sqlbuilder#149]
I usually used Where method like this.

```go
builder.Select().From().Where(checkFilterFunction())
```

And inside of WHERE, the array can have condition or not. Based on the internal functions. If the return of `checkFilterFunction()` is nil, previously there was no `WHERE` in the query. But now, even when the condition is null, it is placing WHERE, and there are no following query, so it causes syntax error on latest version.

When I check the function,

```go
// Where sets expressions of WHERE in SELECT.
func (sb *SelectBuilder) Where(andExpr ...string) *SelectBuilder {
	if sb.WhereClause == nil {
		sb.WhereClause = NewWhereClause()
	}

	sb.WhereClause.AddWhereExpr(sb.args, andExpr...)
	sb.marker = selectMarkerAfterWhere
	return sb
}
```

I think it can be changed like this,

```go
// Where sets expressions of WHERE in SELECT.
func (sb *SelectBuilder) Where(andExpr ...string) *SelectBuilder {
	if andExpr == nil {
		return sb
	}

	if sb.WhereClause == nil {
		sb.WhereClause = NewWhereClause()
	}

	sb.WhereClause.AddWhereExpr(sb.args, andExpr...)
	sb.marker = selectMarkerAfterWhere
	return sb
}
```

Any thought?

**Comment by huandu** [comment]
It should be a bug. I'll fix it. Thanks for reporting this to me.

**Comment by huandu** [comment]
The fix is in tag v1.27.2.

**Comment by wangmir** [comment]
@huandu Thanks for the quick fix!
